ocean_lib/models/v4/test/test_swap_fees.py

```diff
@@ -0,0 +1,3906 @@
+
+    tx = bpool.join_pool(
+        web3.toWei("0.01", "ether"),
+        [
+            web3.toWei("50", "ether"),
+            web3.toWei("50", "ether"),
+        ],
+        publisher_wallet,
+    )
+
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+
+    join_pool_event = bpool.get_event_log(
+        bpool.EVENT_LOG_JOIN,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    assert join_pool_event[0].args.tokenIn == erc20_token.address
+    assert join_pool_event[1].args.tokenIn == dai_contract.address
+
+    assert web3.toWei("0.01", "ether") == bpool.balanceOf(publisher_wallet.address)
+    assert ss_contract_bpt_balance == bpool.balanceOf(
+        get_address_of_type(config, "Staking")
+    )
+    assert ss_contract_dt_balance == erc20_token.balanceOf(
+        get_address_of_type(config, "Staking")
+    )
+
+    # * Tests consumer adds more liquidity with joinswapExternAmountIn (only OCEAN)
+
+    consumer_ocean_balance = dai_contract.balanceOf(consumer_wallet.address)
+    consumer_dt_balance = erc20_token.balanceOf(consumer_wallet.address)
+    ss_contract_dt_balance = erc20_token.balanceOf(side_staking.address)
+    ss_contract_bpt_balance = bpool.balanceOf(side_staking.address)
+
+    dai_contract.approve(bpool_address, web3.toWei(1000, "ether"), consumer_wallet)
+
+    tx = bpool.join_swap_extern_amount_in(
+        dai_contract.address,
+        web3.toWei(1, "ether"),
+        web3.toWei(0.01, "ether"),
+        consumer_wallet,
+    )
+
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    join_pool_event = bpool.get_event_log(
+        bpool.EVENT_LOG_JOIN,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    assert join_pool_event[0].args.tokenIn == dai_contract.address
+    assert join_pool_event[1].args.tokenIn == erc20_token.address
+    assert join_pool_event[0].args.tokenAmountIn == web3.toWei(1, "ether")
+    side_staking_amount_in = ss_contract_dt_balance - erc20_token.balanceOf(
+        side_staking.address
+    )
+
+    ##assert (
+    ##    side_staking.get_data_token_balance(erc20_token.address)
+    ##    == ss_contract_dt_balance - side_staking_amount_in
+    ##)
+
+    assert join_pool_event[1].args.tokenAmountIn == side_staking_amount_in
+
+    assert ss_contract_dt_balance - join_pool_event[
+        1
+    ].args.tokenAmountIn == erc20_token.balanceOf(side_staking.address)
+
+    bpt_event = bpool.get_event_log(
+        bpool.EVENT_LOG_BPT,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+
+    assert bpt_event[0].args.bptAmount + ss_contract_bpt_balance == bpool.balanceOf(
+        side_staking.address
+    )
+    assert erc20_token.balanceOf(consumer_wallet.address) == consumer_dt_balance
+
+    # * Tests consumer adds more liquidity with joinswapPoolAmountOut (only dai)
+
+    consumer_dt_balance = erc20_token.balanceOf(consumer_wallet.address)
```

Comment:
these assignments seem to be repetitive. Can we create a helper function that returns a tuple?